sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -285,11 +285,22 @@ class Analyzer(catalog: Catalog,
             result
 
           // Resolve field names using the resolver.
-          case f @ GetField(child, fieldName) if !f.resolved && child.resolved =>
+          case f @ GetField(child, fieldName) if child.resolved =>
             child.dataType match {
               case StructType(fields) =>
-                val resolvedFieldName = fields.map(_.name).find(resolver(_, fieldName))
-                resolvedFieldName.map(n => f.copy(fieldName = n)).getOrElse(f)
+                val actualField = fields.filter(f => resolver(f.name, fieldName))
+                if (actualField.length == 0) {
+                  sys.error(
+                    s"No such struct field $fieldName in ${fields.map(_.name).mkString(", ")}")
```

Comment:
I think we should have a well defined `resolved` field. Actually, I feel `UnresolvedGetField` is a better idea. I think it is reasonable to have a `UnresolvedXXX` whenever we need to do lookup or name matching to resolve it. @marmbrus What do you think?